### PR TITLE
Add Flower Datasets docs setup

### PR DESCRIPTION
**Pull request by adam-narozniak**
Add the missing element needed for the documentation to be created.

### CODE REVIEW

File: datasets/doc/source/conf.py [hunk 1]
```diff
@@ -26,13 +26,15 @@
 
 
 # Fixing path issue for autodoc
-sys.path.insert(0, os.path.abspath("../../src/py"))
+sys.path.insert(0, os.path.abspath("../../"))
+sys.path.insert(0, os.path.abspath("../../flwr_datasets"))
+sys.path.insert(0, os.path.abspath("../../flwr_datasets/federated_dataset"))
```
Comment (danieljanes) [inline datasets/doc/source/conf.py:31]: Are these two necessary?

File: datasets/doc/source/conf.py [hunk 2]
```diff
@@ -47,6 +49,8 @@
 extensions = [
     "sphinx.ext.napoleon",
     "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
+    "sphinx_autodoc_typehints",
```
Comment (danieljanes) [inline datasets/doc/source/conf.py:53]: Should we add these to other docs projects as well?

Comment (adam-narozniak) [reply datasets/doc/source/conf.py:53]: I use the auto-summary in the project to generate links to classes and methods like here image

Comment (adam-narozniak) [reply datasets/doc/source/conf.py:53]: I will delete the `sphinx_autodoc_typehints`. I didn't finally use it.

File: datasets/doc/source/conf.py (hunk 1 above)
Comment (adam-narozniak) [reply datasets/doc/source/conf.py]: Now, I removed the bottom one, and everything works just fine.

Comment (danieljanes) [reply datasets/doc/source/conf.py]: And how the one ending in `flwr_datasets`? The previous version seems to only point at the directory, not the package itself

Comment (adam-narozniak) [reply datasets/doc/source/conf.py]: When the `sys.path.insert(0, os.path.abspath("../../flwr_datasets"))` is removed then I have as shorter API Reference
image

The full reference:
image

Comment (adam-narozniak) [reply datasets/doc/source/conf.py]: In any cases when I keep just one of them doesn't work (to produce the full API ref). The only setting that works is the 
```
sys.path.insert(0, os.path.abspath("../../"))
sys.path.insert(0, os.path.abspath("../../flwr_datasets"))
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**